tokio-trace-fmt/src/span.rs

```diff
@@ -114,7 +115,10 @@ impl<'a> Span<'a> {
 
 // ===== impl Context =====
 
-impl<'a> Context<'a> {
+impl<'a, N> Context<'a, N>
+where
+    N: ::NewVisitor<'a>,
```

Comment:
I don't think this is the right bound --- the lifetime `NewVisitor` is generic over is the lifetime of the _writer_, not that of the `NewVisitor` itself. I think this needs to be
```suggestion
    N: for<'writer> ::NewVisitor<'writer>,
```

I would give lifetimes descriptive names at this point since it becomes hard to tell them apart

Reply:
Also, I think only the `new_visitor` function itself should have this bound. Getting rid of it on the whole `impl` block would let places where we don't care about the `new_visitor` not require the bound.

Reply:
I dont think this is possible to use the HRTB and then return the associated type :(

```
error[E0212]: cannot extract an associated type from a higher-ranked trait bound in this context
   --> tokio-trace-fmt/src/span.rs:194:78
    |
194 |     pub fn new_visitor(&self, writer: &'a mut fmt::Write, is_empty: bool) -> N::Visitor
    |                                                                              ^^^^^^^^^^
```

Reply:
Sorry, you're right that it doesn't work with the higher-ranked bound, but that isn't actually necessary in this case (since we have a concrete lifetime for the writer here). I think it just needs to be
```rust
    pub fn new_visitor<'writer>(&self, writer: &'writer mut fmt::Write, is_empty: bool) -> N::Visitor
    where
        N: ::NewVisitor<'writer>,
    {
```

The point is that we don't need to require that the writer and the context have the same LT.